docs/site/Inside-Loopback-Application.md

```diff
@@ -0,0 +1,173 @@
+
+Digital transformation is changing how data and services are shared across Web/Mobile/IoT platforms.
+APIs allow easy and secure access to digital resources and capabilities of an organization by internal and enternal parties.
+
+Developers are looking out for more productive ways of creating and consuming APIs.
+
+Consider a typical `Online Shopping` site. First the UI requirements are gathered. Customers would be provided a home page, a product list, a shopping cart and an order history page.
+
+![Shopping Application](imgs/shopping-app.png)
+
+Next, the API requirements are assessed. If the capability to choose products, add to cart and submit an order are exposed as secure APIs, they can be used by an in-house web application as well as a third party mobile app developer.
+
+Once the API requirements are assessed,
+
+- Front end designers would want to try the `Online Shopping` UI experience.
+  > LoopBack provides options to configure simple methods as remote API endpoints.
+
+- API developers would want to assess API specifications and security options.
+  > LoopBack provides options to decorate remote methods with security options.
+
+- Back end developers would want to create Entity-Relationship models from backend datasource
+  > LoopBack provides extensive features to connect to backend and generate object models from datasource schemas.
+
+The example application [loopback4-example-shopping](https://github.com/strongloop/loopback4-example-shopping)
+is a nice place for developers to begin their LoopBack 4 journey.
+
+In the following sections we will see how LoopBack can be used for the online shopping example.
+
+## Use Cases for Online Shopping
+
+For our `Online Shopping` site how the customer might interact with a web or mobile application is the same.
+
+- `Login use cases`
+  - The customer would register as a user.
+  - The customer logs in as a user.
+  - After the user logs in, the mobile/web app creates a shopping cart for the user.
+
+We have the following API requirements for this usecase:
+
+| Use case              | operations         |  summary             | 
+| --------------------  | --------------------- |----------------------|
+| create a new user     | POST /users           | Basic authentication |
+| query for a user      | GET  /users/{userId}  | query by id          |
+| login as a user       | POST /users/login     | returns a jwt token  |
+
+Similar to this we can breakdown further use cases into API requirements.
+
+- `Shopping use cases`
+  - User starts adding and removing items to the shopping cart.
+  - If the user logs out, the shopping cart is saved and used for the next user session.
+
+- `Order use cases`
+  - User creates an order with items in the shopping cart.
+  - The mobile/web app submits the order and resets the shopping cart.
+  - User takes a look at all the previous orders.
+
+![example shopping](https://github.com/strongloop/loopback4-example-shopping/blob/master/example-shopping.png)
```

Comment:
it might just be my preference. Would it be better to have this diagram right under `In the following sections... ` text? It helps to set the stage on what we're going to cover.